examples/dynamic-lora-sidecar/deployment.yaml

```diff
@@ -0,0 +1,129 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: llama-deployment
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: llama-server
+  template:
+    metadata:
+      labels:
+        app: llama-server
+        ai.gke.io/model: LLaMA2_7B
+        ai.gke.io/inference-server: vllm
+        examples.ai.gke.io/source: model-garden
+    spec:
+      shareProcessNamespace: true
+      containers:
+      - name: inference-server
+        image: vllm/vllm-openai:v0.6.3.post1
+        resources:
+          requests:
+            cpu: 5
+            memory: 20Gi
+            ephemeral-storage: 40Gi
+            nvidia.com/gpu : 1
+          limits:
+            cpu: 5
+            memory: 20Gi
+            ephemeral-storage: 40Gi
+            nvidia.com/gpu : 1
+        command: ["/bin/sh", "-c"]
+        args:
+        - vllm serve meta-llama/Llama-2-7b-hf
+        - --host=0.0.0.0
+        - --port=8000
+        - --tensor-parallel-size=1
+        - --swap-space=16
+        - --gpu-memory-utilization=0.95
+        - --max-model-len=2048
+        - --max-num-batched-tokens=4096
+        - --disable-log-stats
+        - --enable-loras
+        - --max-loras=5
+        env:
+        - name: DEPLOY_SOURCE
+          value: UI_NATIVE_MODEL
+        - name: MODEL_ID
+          value: "Llama2-7B"
+        - name: AIP_STORAGE_URI
+          value: "gs://vertex-model-garden-public-us/llama2/llama2-7b-hf"
+        - name: VLLM_ALLOW_RUNTIME_LORA_UPDATING
+          value: "true"
+        - name: HF_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: hf-token  # The name of your Kubernetes Secret
+              key: token   # The specific key within the Secret
+        - name: DYNAMIC_LORA_ROLLOUT_CONFIG
+          value: "/config/configmap.yaml"
+        volumeMounts:
+        - mountPath: /dev/shm
+          name: dshm
+      initContainers:
```

Comment:
why are we setting the sidecar as an initContainer? shouldn't the sidecar run all the time to reconcile on the configmap changes?

Reply:
Do we want to deploy it as normal container in pod with llm server? I am not sure if deployment order needs to be enforced, since model server takes some time to initialize serving  anyways and syncer will wait for it to be healthy.

Reply:
Did you actually test this? I am surprised that it works. initContainers run at the beginning and exit before starting any of the main containers, also we want to actively reconcile the set of adapters to load/unload, and so it should be running all the time, right?

Reply:
ah, I missed the restartPolicy, yes please lets document that.

Reply:
Yes i tested it, its running all the time along with the model server, its not normal initcontainer 😅 . I documented the sidecar use and  the need to enable feature gate prior to 1.29. Its enabled by default from 1.29. There shouldn't be too many prior to 1.29 users. 1.28 is already out of support and before 1.28 there is no official sidecar implementation, added this information to README.